query/src/main/scala/filodb/query/exec/ExecPlan.scala

```diff
@@ -368,10 +372,18 @@ abstract class NonLeafExecPlan extends ExecPlan {
                      (implicit sched: Scheduler): ExecResult = {
     val parentSpan = Kamon.currentSpan()
     parentSpan.mark("create-child-tasks")
+
+    // whether child tasks need to be executed sequentially.
+    // parallelism 1 means, only one worker thread to process underlying tasks.
+    val parallelism: Int = if (parallelChildTasks)
+                              Runtime.getRuntime.availableProcessors()
+                           else
+                              1
+
     // Create tasks for all results.
     // NOTE: It's really important to preserve the "index" of the child task, as joins depend on it
     val childTasks = Observable.fromIterable(children.zipWithIndex)
-                               .mapAsync(Runtime.getRuntime.availableProcessors()) { case (plan, i) =>
+                               .mapAsync(parallelism) { case (plan, i) =>
```

Comment:
Lets discuss this in person - I am still not sure if it will work as we want it to.